drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -282,20 +282,23 @@ void RasterizerSceneGLES3::_geometry_instance_add_surface_with_material(Geometry
 	if (p_material->shader_data->uses_tangent && !(s->format & RS::ARRAY_FORMAT_TANGENT)) {
 		WARN_PRINT_ED("Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool).");
 	}
+	sdcache->pass_index = p_material->pass_index;
 }
 
 void RasterizerSceneGLES3::_geometry_instance_add_surface_with_material_chain(GeometryInstanceGLES3 *ginstance, uint32_t p_surface, GLES3::SceneMaterialData *p_material_data, RID p_mat_src, RID p_mesh) {
 	GLES3::SceneMaterialData *material_data = p_material_data;
 	GLES3::MaterialStorage *material_storage = GLES3::MaterialStorage::get_singleton();
 
 	_geometry_instance_add_surface_with_material(ginstance, p_surface, material_data, p_mat_src.get_local_index(), material_storage->material_get_shader_id(p_mat_src), p_mesh);
-
+	uint64_t pass_index = material_data->pass_index;
```

Comment:
```suggestion
	uint8_t pass_index = material_data->pass_index;
```

I think this should be uint8_ts everywhere no?